File: Python/Excel Sheet Column Title/Excel Sheet Column Title.py
class Solution:
    def convertToTitle(self, n: int) -> str:
        s = ''
        while(n > 0):
            res = n % 26
            n = int((n-res)/26)
            if(n != 0 and res == 0):
                n -= 1
                res = 26
            if(res != 0):
                s += chr(res+64)
            if(n > 0 and n <= 26):
                s += chr(n+64)
                n = 0
        return s[::-1]

File: C#/Excel Sheet Column Title/Excel Sheet Column Title.cs
public class Solution {
    public char NumToChar (int num) {
        return (char) (num + 64);
    }

    public string ConvertToTitle (int n) {
        Stack<char> stk = new Stack<char> ();
        int res = 0, nums = n;
        while (nums > 0) {
            res = nums % 26;
            nums = (nums - res) / 26;
            if (res == 0 && nums != 0) {
                res = 26;
                nums--;
            }
            if (res != 0) stk.Push (NumToChar (res));
            if (nums <= 26 && nums > 0) {
                stk.Push (NumToChar (nums));
                nums = 0;
            }
        }
        StringBuilder sb = new StringBuilder ();
        while (stk.Count > 0) sb.Append (stk.Pop ());
        return sb.ToString ();
    }
}